doc/directives.md

```diff
@@ -0,0 +1,117 @@
+Dartdoc supports several **directives** within Dart doc comments. Each directive
+is then processed during documentation generation, and new text is inserted in
+place of the directive. Not all directives are supported for package
+documentation at https://pub.dev/.
+
+A doc comment directive consists of either one directive tag, which looks
+something like `{@DIRECTIVE ARG=VALUE ...}`, or two such tags (an opening and a
+closing tag), with content in between.
+
+The supported directives are listed below:
+
+## `@nodoc` - Do not include documentation
+
+An element whose doc comment should not appear in the generated documenation can
+include the `@nodoc` directive.
+
+## `{@category}` and `{@subCategory}` - Categories
+
+Elements such as libraries and classes can be grouped into categories and
+sub-categories by adding `{@category CATEGORY NAME}` and
+`{@subCategory SUB-CATEGORY NAME}` in doc comments. Each category then gets its
+own documentation page, listing all of the categorized elements.
```

Comment:
We should consider coming back at some point and removing / generalizing descriptions of the output so that this doesn't need to change every time we change the render engine (especially if we separate the render engine from this package).

Reply:
Ah that's a good point. I think it is important to tell the user what the end result of creating and using categories is. But I think you're right: the text here should be a little broad, and a separate doc should flesh out the concept of categories and how they are displayed and sorted, etc.